src/audio/component.c

```diff
@@ -29,31 +29,45 @@ DECLARE_SOF_UUID("component", comp_uuid, 0x7c42ce8b, 0x0108, 0x43d0,
 
 DECLARE_TR_CTX(comp_tr, SOF_UUID(comp_uuid), LOG_LEVEL_INFO);
 
-static const struct comp_driver *get_drv(uint32_t type)
+static const struct comp_driver *get_drv(struct sof_ipc_comp *comp)
 {
 	struct comp_driver_list *drivers = comp_drivers_get();
 	struct list_item *clist;
 	const struct comp_driver *drv = NULL;
 	struct comp_driver_info *info;
-	uint32_t flags;
-
-	irq_local_disable(flags);
 
-	/* search driver list for driver type */
-	list_for_item(clist, &drivers->list) {
-		info = container_of(clist, struct comp_driver_info, list);
-		if (info->drv->type == type) {
-			drv = info->drv;
-			platform_shared_commit(info, sizeof(*info));
-			goto out;
+	/* use UUID to match the driver if UUID is provided */
+	if (sof_is_uuid_nil(comp->uuid)) {
```

Comment:
Not sure I follow the logic.

this looks inverted, no? should this be if (!sof_is_uuid_nil()) ?

Reply:
You are right, just focused adding '!' on the memcmp below and forgot the one here, thanks.